gixy/plugins/proxy_pass_normalized.py

```diff
@@ -0,0 +1,44 @@
+import re
+import gixy
+from gixy.plugins.plugin import Plugin
+
+class ProxyPassDecodedRegex(Plugin):
+    r"""
+    This plugin detects if there is any path component (slash or more)
+    after the host in a proxy_pass directive.
+    Example flagged directives:
+        proxy_pass http://backend/;
+        proxy_pass http://backend/foo/bar;
+    """
+
+    summary = 'Detect path after host in proxy_pass (potential URL decoding issue)'
+    severity = gixy.severity.LOW
+    description = ("A slash immediately after the host in proxy_pass leads to the path being decoded and normalized before proxying downstream, leading to unexpected behavior related to encoded slashes.")
+    help_url = 'https://joshua.hu/proxy-pass-nginx-decoding-normalizing-url-path-dangerous#nginx-proxy_pass'
+    directives = ['proxy_pass']
+
+    def __init__(self, config):
+        super(ProxyPassDecodedRegex, self).__init__(config)
+        self.parse_uri_re = re.compile(r'(?P<scheme>[^?#/)]+://)?(?P<host>[^?#/)]+)(?P<path>/.*)?')
```

Comment:
Looks like this regex is invalid. See https://regex101.com/r/oaTGN8/2
Maybe use a simple URL parsing instead of regex?

Reply:
The regex is valid, you need to include the <groups> in regex101, and use regex version 2.

The regex is partially copied from https://github.com/dvershinin/gixy/blob/master/gixy/plugins/ssrf.py#L32C42-L32C84